src/autopas/pairwiseFunctors/LJFunctor.h

```diff
@@ -653,12 +513,265 @@ class LJFunctor : public Functor<Particle, ParticleCell, typename Particle::SoAA
   }
 
  private:
+  template <bool newton3, bool duplicatedCalculations>
+  void SoAFunctorImpl(SoA<SoAArraysType> &soa,
+                      const std::vector<std::vector<size_t, autopas::AlignedAllocator<size_t>>> &neighborList,
+                      size_t iFrom, size_t iTo) {
+    auto numParts = soa.getNumParticles();
+    AutoPasLog(debug, "SoAFunctorVerlet: {}", soa.getNumParticles());
+
+    if (numParts == 0) return;
+
+    double *const __restrict__ xptr = soa.template begin<Particle::AttributeNames::posX>();
+    double *const __restrict__ yptr = soa.template begin<Particle::AttributeNames::posY>();
+    double *const __restrict__ zptr = soa.template begin<Particle::AttributeNames::posZ>();
+
+    double *const __restrict__ fxptr = soa.template begin<Particle::AttributeNames::forceX>();
+    double *const __restrict__ fyptr = soa.template begin<Particle::AttributeNames::forceY>();
+    double *const __restrict__ fzptr = soa.template begin<Particle::AttributeNames::forceZ>();
+
+    const double cutoffsquare = _cutoffsquare, epsilon24 = _epsilon24, sigmasquare = _sigmasquare, shift6 = _shift6;
+
+    const std::array<double, 3> lowCorner = {_lowCorner[0], _lowCorner[1], _lowCorner[2]};
+    const std::array<double, 3> highCorner = {_highCorner[0], _highCorner[1], _highCorner[2]};
+
+    double upotSum = 0.;
+    double virialSumX = 0.;
+    double virialSumY = 0.;
+    double virialSumZ = 0.;
+
+    for (size_t i = iFrom; i < iTo; ++i) {
+      double fxacc = 0;
+      double fyacc = 0;
+      double fzacc = 0;
+      const size_t listSizeI = neighborList[i].size();
+      const size_t *const __restrict__ currentList = neighborList[i].data();
+
+      // checks whether particle 1 is in the domain box, unused if _duplicatedCalculations is false!
+      bool inbox1;
+      double inbox1Mul = 0.;
+      if (duplicatedCalculations) {  // only for duplicated calculations we need this value
+        inbox1 = autopas::utils::inBox({xptr[i], yptr[i], zptr[i]}, lowCorner, highCorner);
+        inbox1Mul = inbox1 ? 1. : 0.;
+        if (newton3) {
+          inbox1Mul *= 0.5;
+        }
+      }
+
+      // this is a magic number, that should correspond to at least
+      // vectorization width*N have testet multiple sizes:
+      // 4: does not give a speedup, slower than original AoSFunctor
+      // 8: small speedup compared to AoS
+      // 12: highest speedup compared to Aos
+      // 16: smaller speedup
+      // in theory this is a variable, we could auto-tune over...
+#ifdef __AVX512F__
+      // use a multiple of 8 for avx
+      const size_t vecsize = 16;
```

Comment:
this should probably a constexpr for the whole class

Reply:
changed to constexpr, but left where it is.